scripts/protocolFilters/fonbnk.test.ts

```diff
@@ -0,0 +1,177 @@
+import { Log, QueryResponse } from '@envio-dev/hypersync-client'
+import { NetworkId, ReferralEvent } from '../types'
+import { getBlock, getHyperSyncClient } from '../utils'
+import { filter } from './fonbnk'
+import {
+  getFonbnkAssets,
+  getPayoutWallets,
+} from '../calculateRevenue/protocols/fonbnk/helpers'
+import { Address } from 'viem'
+
+jest.mock('../utils', () => ({
+  getHyperSyncClient: jest.fn(),
+  getBlock: jest.fn(),
+}))
+jest.mock('../calculateRevenue/protocols/fonbnk/helpers')
+
+const MOCK_ADDRESS = '0x1234567890123456789012345678901234567890' as Address
+
+const makeQueryResponse = (logs: Log[], nextBlock = 100): QueryResponse => ({
+  data: {
+    blocks: [],
+    transactions: [],
+    logs,
+    traces: [],
+  },
+  nextBlock,
+  totalExecutionTime: 50,
+})
+
+describe('filter', () => {
+  const userAddress = MOCK_ADDRESS
+  const event: ReferralEvent = {
+    userAddress: userAddress,
+    timestamp: Math.round(new Date('2025-03-18T00:00:00Z').getTime() / 1000),
+    protocol: 'aerodrome',
+    referrerId: 'referrer1',
+  }
+  let mockClient: { get: jest.Mock }
+
+  beforeEach(() => {
+    jest.clearAllMocks()
+    mockClient = { get: jest.fn() }
+    jest
+      .mocked(getHyperSyncClient)
+      .mockReturnValue(
+        mockClient as unknown as ReturnType<typeof getHyperSyncClient>,
+      )
+    jest
+      .mocked(getFonbnkAssets)
+      .mockResolvedValue([{ network: 'CELO', asset: 'USDC' }])
+    jest.mocked(getPayoutWallets).mockResolvedValue(['0x123'])
+  })
+
+  it('returns false if first found block is before the referral event timestamp', async () => {
+    mockClient.get.mockResolvedValueOnce(
+      makeQueryResponse([{ blockNumber: 123, topics: [] }]),
+    )
+
+    jest.mocked(getBlock).mockImplementation(
+      (_networkId: NetworkId, _blockNumber: bigint) =>
+        Promise.resolve({
+          timestamp: 1742194800n, // March 17, 2025
+        }) as unknown as ReturnType<typeof getBlock>,
+    )
+
+    const result = await filter(event)
+    expect(result).toBe(false)
+    expect(mockClient.get).toHaveBeenCalledTimes(1)
+    expect(getBlock).toHaveBeenCalled()
```

Comment:
nit: if we want the tests to provide good coverage for how `getBlock` is called, we can either do `expect(getBlock).toHaveBeenCalledWith(...)` or in the mock implementation, throw an error if the networkId / blockNumber are not what we expect. with the current tests, i could accidentally leave a hard coded value in the input arg of `getBlock` (from debugging, idk) and the tests would not catch that